common/src/main/java/org/mcaccess/minecraftaccess/mixin/LivingEntityMixin.java

```diff
@@ -14,29 +15,20 @@
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
 
 import org.mcaccess.minecraftaccess.features.EffectNarrator;
-import org.mcaccess.minecraftaccess.utils.WorldUtils;
 
 @SuppressWarnings("EqualsBetweenInconvertibleTypes")
 @Mixin(LivingEntity.class)
 abstract class LivingEntityMixin {
-
-    @Inject(method = "onEffectUpdated", at = @At("TAIL"))
-    public void narrateEffectApplication(MobEffectInstance effectInstance, boolean forced, Entity entity, CallbackInfo ci) {
-        if (Objects.equals(WorldUtils.getClientPlayer(), this)) {
-            EffectNarrator.narrateGained(effectInstance);
-        }
-    }
-
-    @Inject(method = "onEffectAdded", at = @At("TAIL"))
-    public void narrateEffectApplication2(MobEffectInstance effectInstance, Entity entity, CallbackInfo ci) {
-        if (Objects.equals(WorldUtils.getClientPlayer(), this)) {
+    @Inject(method = {"onEffectUpdated", "onEffectAdded"}, at = @At("TAIL"))
+    private void narrateEffectApplication(MobEffectInstance effectInstance, boolean forced, Entity entity, CallbackInfo ci) {
```

Comment:
`onEffectAdded` doesn't have a `forced` parameter and from a quick two minutes of experimentation with the export thingo just now it looks like if you have parameters and they don't perfectly match Mixin will *silently* not inject (amazing) even if your parameters are a subset of the actual method, but not having any parameters other than `CallbackInfo` is perfectly valid and will inject according to the JavaDoc of `@Inject`. I wonder if Mixin Extras has anything more powerful?